src/libraries/filcrypto/filproofs/algorithms.go

```diff
@@ -879,9 +880,14 @@ func zeroPadding(size UInt) PieceInfo {
 
 func joinPieceInfos(left PieceInfo, right PieceInfo) PieceInfo {
 	util.Assert(left.Size() == right.Size())
+
+	// FIXME: make this whole function generic?
+	sectorPieceCID, err := cid.Cast(BinaryHash_SHA256Hash(cid.Cid(left.PieceCID()).Bytes(), cid.Cid(right.PieceCID()).Bytes()))
```

Comment:
This is slightly more complicated, see [this discussion](https://filecoinproject.slack.com/archives/CHMNDCK9P/p1578629688082700)

We don't need to fully implement it here, but should make a note that, e.g. `cid.Bytes()` isn't actually the payload data that we want input to the binary hash function. Leaving a comment should suffice. FYI @porcuquine in case you want to specify what we leave here.